### PR TITLE
Rename types so they differ from inferred types

### PR DESCRIPTION
**Pull request by dbrattli**

### CODE REVIEW

File: src/Builder.fs [hunk 1]
```diff
@@ -27,9 +27,9 @@ type RequestBuilder () =
         |> Seq.map func
         |> sequential
 
-    member _.Bind(source: HttpHandler<'a, 'b, 'r, 'err>, fn: 'b -> HttpHandler<'a, 'c, 'r, 'err>) : HttpHandler<'a, 'c, 'r, 'err> =
+    member _.Bind(source: HttpHandler<'T, 'T2, 'TResult, 'TError>, fn: 'T2 -> HttpHandler<'T, 'T3, 'TResult, 'TError>) : HttpHandler<'T, 'T3, 'TResult, 'TError> =
```
Comment (exyi) [inline src/Builder.fs]: I think it would be a little bit better to call `T2` `TNext` as in `HttpHandler<'T, 'TNext, 'TResult, 'TError>`

Comment (dbrattli) [reply src/Builder.fs]: But there is a T3 as well. TNext1, TNext2?

Comment (exyi) [reply src/Builder.fs]: Aha, I didn't notice. `TNext1`, `TNext2` would be the best, IMHO.

File: benchmark/Common.fs [hunk 2]
```diff
@@ -112,7 +112,7 @@ let readNewtonsoft<'a> (stream: IO.Stream) =
         | e -> return Error (e.ToString())
     }
 
-let jsonReader<'a, 'r, 'err> (reader: Stream -> Task<Result<'a, string>>) (next: NextFunc<'a,'r, 'err>) (context: HttpContext) : HttpFuncResult<'r, 'err> =
+let jsonReader<'a, 'r, 'err> (reader: Stream -> Task<Result<'a, string>>) (next: HttpFunc<'a,'r, 'err>) (context: HttpContext) : HttpFuncResult<'r, 'err> =
```
Comment (wjoel) [inline benchmark/Common.fs]: Should this also be changed to 'T, 'TResult, 'TError?

Comment (dbrattli) [reply benchmark/Common.fs]: Yes, haven't fixed all the testing code, benchmarks yet ...

**Comment by dbrattli** [comment]
Thanks for reviewing

**Pull request merged**